Packs/ProofpointServerProtection/Integrations/ProofpointProtectionServerV2/ProofpointProtectionServerV2.py

```diff
@@ -0,0 +1,316 @@
+    ))
+    return CommandResults(readable_output=result)
+
+
+def move_message(client: Client, args: Dict[str, Any]) -> CommandResults:
+    local_guid = args.get('local_guid')
+    result = client.quarantine_action_request(
+        action='move',
+        folder=args.get('folder_name'),
+        localguid=local_guid,
+        targetfolder=args.get('target_folder'),
+    )
+    if isinstance(result, dict):
+        return CommandResults(readable_output=result.get('status', f'Successfully moved message {local_guid}'))
+    raise RuntimeError(f'Message move action failed.\n{result}')
+
+
+def delete_message(client: Client, args: Dict[str, Any]) -> CommandResults:
+    local_guid = args.get('local_guid')
+    result = client.quarantine_action_request(
+        action='delete',
+        folder=args.get('folder_name'),
+        localguid=local_guid,
+        deletedfolder=args.get('deleted_folder'),
+    )
+    if isinstance(result, dict):
+        return CommandResults(readable_output=result.get('status', f'Successfully deleted message {local_guid}'))
+    raise RuntimeError(f'Message delete action failed.\n{result}')
+
+
+def download_message(client: Client, args: Dict[str, Any]) -> Union[CommandResults, Dict]:
+    guid = args.get('guid', '')
+    result = client.download_message_request(guid)
+    if result.status_code == 404:
+        return CommandResults(readable_output='No message found.')
+    return fileResult(guid + '.eml', result.content)
+
+
+def main() -> None:
+    try:
+        command = demisto.command()
+        params = demisto.params()
+        client = Client(
+            base_url=urljoin(params['url'], '/rest/v1'),
+            auth=(params['credentials']['identifier'], params['credentials']['password']),
+            verify=not params.get('insecure', False),
+            proxy=params.get('proxy', False),
+        )
+        if command == 'test-module':
+            return_results(test_module(client))
+        elif command == 'proofpoint-pps-smart-search':
+            return_results(smart_search(client, demisto.args()))
+        elif command == 'proofpoint-pps-list-quarantined-messages':
+            return_results(list_quarantined_messages(client, demisto.args()))
+        elif command == 'proofpoint-pps-release-message':
+            return_results(release_message(client, demisto.args()))
+        elif command == 'proofpoint-pps-resubmit-message':
+            return_results(resubmit_message(client, demisto.args()))
+        elif command == 'proofpoint-pps-forward-message':
+            return_results(forward_message(client, demisto.args()))
+        elif command == 'proofpoint-pps-move-message':
+            return_results(move_message(client, demisto.args()))
+        elif command == 'proofpoint-pps-delete-message':
+            return_results(delete_message(client, demisto.args()))
+        elif command == 'proofpoint-pps-download-message':
+            return_results(download_message(client, demisto.args()))
+
```

Comment:
add `else` with `NotImplementedError` (new convention)

Reply:
isnt just for mirroring ?